### PR TITLE
Upgrade to ecalir v0.3.2

### PR DESCRIPTION
**Pull request by rorp**
Note that /allupdates API is broken in Eclair 0.3.2 (https://github.com/ACINQ/eclair/issues/1179), so all test cases that use it are marked as pending.

### CODE REVIEW

File: eclair-rpc/src/main/scala/org/bitcoins/eclair/rpc/json/EclairModels.scala [hunk 1]
```diff
@@ -136,14 +143,20 @@ case class RelayedPayment(
 )
 
 case class SentPayment(
-    amount: MilliSatoshis,
-    feesPaid: MilliSatoshis,
     paymentHash: Sha256Digest,
-    paymentPreimage: String,
-    toChannelId: FundedChannelId,
-    timestamp: FiniteDuration
+    paymentPreimage: String
```
Comment (Christewart) [inline eclair-rpc/src/main/scala/org/bitcoins/eclair/rpc/json/EclairModels.scala]: `paymentPreimage: PaymentPreimage`

File: eclair-rpc/src/main/scala/org/bitcoins/eclair/rpc/json/EclairModels.scala [hunk 2]
```diff
@@ -136,14 +143,20 @@ case class RelayedPayment(
...
 )
 
+object SentPayment {
+  case class Part(
+      id: String,
```
Comment (Christewart) [inline eclair-rpc/src/main/scala/org/bitcoins/eclair/rpc/json/EclairModels.scala]: should this be `id: UUID` ?

File: eclair-rpc/src/main/scala/org/bitcoins/eclair/rpc/json/EclairModels.scala [hunk 3]
```diff
@@ -290,48 +303,63 @@ case class InvoiceResult(
     paymentHash: Sha256Digest,
     expiry: FiniteDuration)
 
+case class PaymentId(value: String) {
```
Comment (Christewart) [inline eclair-rpc/src/main/scala/org/bitcoins/eclair/rpc/json/EclairModels.scala]: Can we tighten the type here? Is this a `UUID`?

File: eclair-rpc/src/main/scala/org/bitcoins/eclair/rpc/json/EclairModels.scala [hunk 4]
```diff
@@ -290,48 +303,63 @@ case class InvoiceResult(
...
-    completedAt: Option[FiniteDuration],
+    paymentRequest: Option[PaymentRequest],
     status: PaymentStatus)
 
 case class ReceivedPaymentResult(
-    paymentHash: Sha256Digest,
-    amountMsat: MilliSatoshis,
-    receivedAt: FiniteDuration)
+    paymentRequest: Option[PaymentRequest],
+    paymentPreimage: String,
```
Comment (Christewart) [inline eclair-rpc/src/main/scala/org/bitcoins/eclair/rpc/json/EclairModels.scala]: `paymentPreimage: PaymentPreimage`

File: eclair-rpc/src/main/scala/org/bitcoins/eclair/rpc/json/EclairModels.scala [hunk 5]
```diff
@@ -290,48 +303,63 @@ case class InvoiceResult(
...
-    signature: LnInvoiceSignature)
+    serialized: String,
+    description: String,
+    paymentHash: Sha256Digest,
+    expiry: FiniteDuration,
+    amount: Option[MilliSatoshis])
 
 case class PaymentResult(
     id: String,
+    parentId: String,
```
Comment (Christewart) [inline eclair-rpc/src/main/scala/org/bitcoins/eclair/rpc/json/EclairModels.scala]: both of these can be `UUID` i believe

File: eclair-rpc/src/main/scala/org/bitcoins/eclair/rpc/json/EclairModels.scala [hunk 6]
```diff
@@ -290,48 +303,63 @@ case class InvoiceResult(
...
+    extends PaymentStatus
+
+case class Hop(
+    nodeId: NodeId,
+    nextNodeId: NodeId,
+    shortChannelId: ShortChannelId)
+
+case class PaymentFailed(failures: Seq[PaymentFailure]) extends PaymentStatus
+
+case class PaymentFailureType(name: String)
```
Comment (Christewart) [inline eclair-rpc/src/main/scala/org/bitcoins/eclair/rpc/json/EclairModels.scala]: How many different `PaymentFailureType` is there? We should probably enumerate them with an ADT so we handle these in a more robust way.

File: testkit/src/main/scala/org/bitcoins/testkit/eclair/rpc/EclairRpcTestUtil.scala [hunk 7]
```diff
@@ -255,7 +263,7 @@ trait EclairRpcTestUtil extends BitcoinSLogger {
       failFast: Boolean = true)(implicit system: ActorSystem): Future[Unit] = {
     awaitUntilPaymentStatus(client,
                             paymentId,
-                            PaymentStatus.SUCCEEDED,
+                            "sent",
```
Comment (Christewart) [inline testkit/src/main/scala/org/bitcoins/testkit/eclair/rpc/EclairRpcTestUtil.scala]: I don't like removing types here and replacing them with `String`. Now we will never get compiler errors if these things are changed again. Why did you remove the `PaymentStatus` here?

File: eclair-rpc/src/main/scala/org/bitcoins/eclair/rpc/json/EclairModels.scala [hunk 8]
```diff
@@ -290,48 +303,63 @@ case class InvoiceResult(
...
-  def apply(s: String): PaymentStatus = s match {
-    case "PENDING"   => PENDING
-    case "SUCCEEDED" => SUCCEEDED
-    case "FAILED"    => FAILED
-    case err =>
-      throw new IllegalArgumentException(
-        s"Unknown payment status code `${err}`")
-  }
-}
+case object PaymentPending extends PaymentStatus
```
Comment (Christewart) [inline eclair-rpc/src/main/scala/org/bitcoins/eclair/rpc/json/EclairModels.scala]: Why did we get rid of `PaymentStatus`?

File: testkit/src/main/scala/org/bitcoins/testkit/eclair/rpc/EclairRpcTestUtil.scala (hunk 7 above)
Comment (rorp) [reply testkit/src/main/scala/org/bitcoins/testkit/eclair/rpc/EclairRpcTestUtil.scala]: Because there's no such entity in the RPC responses anymore

File: eclair-rpc/src/main/scala/org/bitcoins/eclair/rpc/json/EclairModels.scala (hunk 8 above)
Comment (rorp) [reply eclair-rpc/src/main/scala/org/bitcoins/eclair/rpc/json/EclairModels.scala]: Because this RPC was changed significantly.

File: eclair-rpc/src/main/scala/org/bitcoins/eclair/rpc/json/EclairModels.scala (hunk 2 above)
Comment (rorp) [reply eclair-rpc/src/main/scala/org/bitcoins/eclair/rpc/json/EclairModels.scala]: Why?

File: testkit/src/main/scala/org/bitcoins/testkit/eclair/rpc/EclairRpcTestUtil.scala (hunk 7 above)
Comment (Christewart) [reply testkit/src/main/scala/org/bitcoins/testkit/eclair/rpc/EclairRpcTestUtil.scala]: So do we need to change the `awaitUntilPaymentStatus` interface to account for this then? 

Could we keep the `PaymentStatus` stuff, but make it internal to `EclairRpcTestUtil` and make it `private`?

Comment (rorp) [reply testkit/src/main/scala/org/bitcoins/testkit/eclair/rpc/EclairRpcTestUtil.scala]: Those string  literals are only used locally inside `EclairRpcTestUtil.awaitUntilPaymentStatus` I cant see any problem with doing it. I can replace them with `Int`s.

Comment (Christewart) [reply testkit/src/main/scala/org/bitcoins/testkit/eclair/rpc/EclairRpcTestUtil.scala]: Fair enough, as long as we don't expose `awaitUntilPaymentStatus()`. I don't like using raw Strings as parameters as it is very hard to update  as we don't get compiler errors if we add a new String that represents a state. For instance if we had to segregate "failed" into "failed-onchain"/"failed-routing". We need to remember to update it, rather than having the compiler work for us and remind us to update it.

File: eclair-rpc/src/main/scala/org/bitcoins/eclair/rpc/json/EclairModels.scala (hunk 2 above)
Comment (Christewart) [reply eclair-rpc/src/main/scala/org/bitcoins/eclair/rpc/json/EclairModels.scala]: It seems that every other `id` in eclair is a UUID.

However I deleted this `object SentPayment` in src and it is only used in one other place -- json serializers. Why do we have this?
